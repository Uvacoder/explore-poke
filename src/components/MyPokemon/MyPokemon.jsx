import { useEffect, useState } from "react";
import { useRouter } from "next/router";
import { MY_POKEMON } from "src/constants";
import { EmptyPokemon, ButtonCatchPokemon, MyPokemonCardWrapper, ReleaseButton }
  from "src/components/MyPokemon/MyPokemonStyled";
import PokemonList from "src/components/PokemonList/PokemonList";

export default function MyPokemon() {
  const [myPokemon, setMyPokemon] = useState([]);
  const router = useRouter();

  useEffect(() => {
    const myPokemonData = JSON.parse(window.localStorage.getItem(MY_POKEMON));
    const checkPokemon = myPokemonData ?? [];
    const transformPokemonData = checkPokemon.map((pokemon) => {
      const checkImage = pokemon.sprites ? pokemon.sprites.front_default : pokemon.image;
      return {
        id: pokemon.id,
        name: pokemon.name,
        newName: pokemon.newName,
        image: checkImage
      }
    });
    setMyPokemon(transformPokemonData);
  }, []);

  const redirectToDetail = (pokemon) => {
    router.push(`pokemon/${pokemon.name}`);
  }

  const releasePokemon = (id) => {
    const confirmRelease = window.confirm('Are you sure you want to release Pokemon?');
    if (confirmRelease) {
      const release = myPokemon.filter(pokemon => pokemon.id !== id);
      setMyPokemon(release);
      window.localStorage.setItem(MY_POKEMON, JSON.stringify(release));
    } else {
      return;
    }
  }

  return (
    <>
      {myPokemon.length ? (
        <MyPokemonCardWrapper>
          {myPokemon.map(pokemon => (
            <PokemonList
              key={pokemon.id+`${Math.random()}`}
              name={pokemon.newName}
              image={pokemon.image}
              showOwned={false}
              handleClick={() => redirectToDetail(pokemon)}
            >
              <ReleaseButton onClick={() => releasePokemon(pokemon.id)}>
                Release
              </ReleaseButton>
            </PokemonList>
          ))}
        </MyPokemonCardWrapper>
      ) : (
        <EmptyPokemon>
          You haven`t caught a pokemon yet!
          <ButtonCatchPokemon onClick={() => router.push('/')}>
            catch it now!
          </ButtonCatchPokemon>
        </EmptyPokemon>
      )}
    </>
  )
}