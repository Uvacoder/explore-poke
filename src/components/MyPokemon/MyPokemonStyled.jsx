import styled from "@emotion/styled";

export const MyPokemonWrapper = styled.div`
  min-height: 90vh;
  width: 100%;
  margin: 0 auto;
  min-height: 100vh;
  @media screen and (max-width: 460px) {
    margin: 20px auto 60px;
    width: 70%;
  }
`

export const EmptyPokemon = styled.h2`
  text-align: center;
  margin-top: 100px;
  display: flex;
  flex-direction: column;
`

export const ButtonCatchPokemon = styled.button`
  margin: 20px auto;
  border: none;
  border-radius: 8px;
  background: #f26257;
  cursor: pointer;
  font-weight: bold;
  padding: 10px 20px;
  color: #fff;
  &:hover {
    background: #f77268;
  }
`

export const MyPokemonCardWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  margin: 0px auto;
`

export const MyPokemonCard = styled.div`
  width: 100%;
  height: auto;
  color: #fff;
  background-color: ${(props) => props.color ? props.color : 'black' };
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 10px 10px 5px 10px;
  border-radius: 5px;
  cursor: pointer;
  @media screen and (min-width: 900px) {
    position: relative;
    top: 25px;
  }
  @media screen and (min-width: 450px) {
    flex: 25%;
    position: relative;
    top: 35px;
  }
  @media (min-width: 400px) and (max-width: 500px) {
    flex: 50%;
    position: relative;
    top: 35px;
  }
`

export const MyPokemonName = styled.h2`
  margin-top: 0px;
  margin-right: 5px;
  color: #fff;
  text-transform: capitalize;
  @media screen and (min-width: 450px) {
    font-size: 20px;
  }
  @media (min-width: 400px) and (max-width: 500px) {
    font-size: 25px;
  }
`

export const PokemonBox = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 100%;
`

export const ReleaseButton = styled(ButtonCatchPokemon)`

`


