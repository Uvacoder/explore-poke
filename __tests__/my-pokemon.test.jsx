import { render } from "@testing-library/react";
import MyPokemon from "pages/my-pokemon";

describe("MyPokemon Home Page", () => {
  it('should render my pokemon without crash', () => {
    const { container } = render(
      <MyPokemon />
    );
    expect(container).not.toBeNull();
  });

  it('should render my pokemon lists', () => {
    const { container } = render(
      <MyPokemon />
    );
    expect(container).toMatchSnapshot();
  });
});