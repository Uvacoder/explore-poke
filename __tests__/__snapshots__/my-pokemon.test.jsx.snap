// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Home Page should render my pokemon lists 1`] = `
<div>
  <div
    class="css-1kunwhw"
  >
    <h2
      class="css-1bz0bai"
    >
      You haven\`t caught a pokemon yet!
      <button
        class="css-ltufum"
      >
        catch it now!
      </button>
    </h2>
  </div>
</div>
`;
